lazyflow/utility/io_util/OMEZarrStore.py

```diff
@@ -256,13 +257,42 @@ def _get_zarr_cache_max_size() -> int:
     return math.floor(caches_max * permissible_fraction_max)
 
 
-def _unescape_and_get_store(uri: str, mode="r", **kwargs) -> FSStore:
+def _ensure_connection_and_get_store(uri: str, mode="r", **kwargs) -> FSStore:
     if uri.startswith("file:"):
         # Zarr's FSStore implementation doesn't unescape file URLs before piping them to
         # the file system. We do it here the same way as in pathHelpers.uri_to_Path.
         # Primarily this is to deal with spaces in Windows paths (encoded as %20).
         uri = os.fsdecode(unquote_to_bytes(uri))
-    return FSStore(uri, mode=mode, **kwargs)
+    store = FSStore(uri, mode=mode, **kwargs)
+    try:
+        store["test"]
+        # Any error not handled here is either a successful connection (404 for "test"), or an unknown problem
+    except KeyError as e:
+        # FSStore wraps some errors in KeyError (FileNotFoundError even double-wrapped)
+        if isinstance(e.__context__, ClientConnectorError):
+            raise ConnectionError(f"Could not connect to {e.__context__.host}:{e.__context__.port}.") from e
+    except ClientResponseError as e:
+        if not (e.status == 403 and uri.startswith("https://")):
+            raise
+        # 403 indicates we need authentication; FSStore maps https: to HTTPFileSystem, which cannot authenticate
```

Comment:
it would be nice to give some sort of feedback to the user that you are trying to authenticate (and how) in the logs at least